### DISCUSSION TITLE
Dynamic Schematron-Validation and BT-71-Lot

### DISCUSSION THREAD

**Discussion by dnl50** [OP-TED/eForms-SDK#544] [Q&A] [answered]
I don't know if this is an issue just yet, so I will open a discussion about this:

When applying the dynamic Schematron-Rules for a Change Notice (Subtype Code 20) **without** `BT-71-Lot`, the Schematron Report always contains the following message:

_The BT-71 Reserved Participation for Lot LOT-XXXX  must be the same as for the parent notice._

After some investigation I discovered that this Message can be printed by two separate rule, which use different XPath-Expression to compare the values between the parent and the child notice.

The first one uses the XPath `deep-equal` function (see [dynamic-validation-stage-6b.sch Line 213](https://github.com/OP-TED/eForms-SDK/blob/1.8.0-rc.2/schematrons/dynamic/validation-stage-6b.sch#L213)) and the other one just compares the two values with the `=` operator (see [dynamic-validation-stage-6b.sch Line 264](https://github.com/OP-TED/eForms-SDK/blob/1.8.0-rc.2/schematrons/dynamic/validation-stage-6b.sch#L264)). The rule using the `deep-equal` function works as expected, but the one using the `=` operator is triggered, eventhough the XPath Expression does not match any nodes.

When the value of `BT-71-Lot` actually changes, the validation report contains the message mentioned above two times. Is this a problem with the SDK?

**Comment by bertrand-lorentz** [comment]
Thanks for your feedback !

We indeed changed the rules in dynamic-validation-stage-6b for this kind of problem:
https://github.com/OP-TED/eForms-SDK/commit/b37b83510bbdd976f462de8a0c1604f6c85385a9

The one you mention might have slipped through. We'll have a closer look at it.

**Comment by bertrand-lorentz** [comment, marked as answer]
In the meantime, the list of fields that must have the same value in a change notice and in the original notice has been significantly reduced.
We've reviewed the content of "validation-stage-6b.sch" so that it does not contradict this. This meant removing a significant number of the rules, including both rules for `BT-71-Lot`. So the problem described here is resolved by this change, which will be available in SDK 1.9.0.

The commit with the change is:
https://github.com/OP-TED/eForms-SDK/commit/8d6d8e7d9b238551108b2f8601e4080307d116aa

Please note that not all rules on change notices are currently implemented in "validation-stage-6b.sch". We are working on a different approach for those rules, so that we don't have to maintain them entirely manually.
